pilot/pkg/model/service.go

```diff
@@ -1035,14 +1041,36 @@ func (i ServiceInfo) GetConditions() ConditionSet {
 			},
 		}
 	} else if i.Waypoint.Error != nil {
+		// if there is no waypoint name, the error will have to do with the waypoint itself, not
+		// the ingress waypoint label.
 		set[WaypointBound] = []Condition{
 			{
 				Status:  false,
 				Reason:  i.Waypoint.Error.Reason,
 				Message: i.Waypoint.Error.Message,
 			},
 		}
+		writeError = false
 	}
+
+	if i.Waypoint.IngressUseWaypoint {
+		set[IngressUsingWaypoint] = []Condition{
+			{
+				Status:  true,
+				Reason:  string(WaypointAccepted),
+				Message: fmt.Sprintf("Successfully attached to waypoint %v. Ingress traffic will traverse the waypoint", i.Waypoint.ResourceName),
+			},
+		}
+	} else if writeError && i.Waypoint.Error != nil {
```

Comment:
do we actually want to write a full second "attached to waypoint" condition with ingress note added on? If we want a second condition maybe we don't repeat the verbiage of the waypoint bound condition and just say, "Ingress traffic will traverse waypoint <name>". I think we could just have 1 condition and append the note about ingress use if it's applicable.